### PR TITLE
content filter renamed and moved to sidebar

### PR DESCRIPTION
**Pull request by sarahkeh**
## Description

This ended up requiring some more ToggleButtonGroup work, but these changes cover:

1. Removed the SuperAdmin filter and moved it to the DrawerRail
2. Removes any references to Super Admin, renamed to ContentFilterControls
3. Realized the new theme didn't quite match the mock so I added a new variant to ToggleButtonGroup called containedLight.  I also added a centerIcon prop like the MainButton uses to improve the icon styling a bit.  These two additions can best tested in the Sandbox
4. Added two new icons, ShieldOffIcon and ShieldCheckIcon
5. Tooltip added to buttons
6. Default state set to off

References: CV2-5828

## Type of change

- [ ] Performance improvement and/or refactoring (non-breaking change that keeps existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Security mitigation or enhancement
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Automated test (add or update automated tests)

## How has this been tested?

Style changes have been tested in the Sandbox, verified the toggle works anywhere in the app. 

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I've made sure my branch is runnable and given good testing steps in the PR description
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I implemented any new components, they are self-contained, their `propTypes` are declared and they use React Hooks and, if data-fetching is required, they use Relay Modern with fragment containers
- [ ] If my components involve user interaction - specifically button, text fields, or other inputs - I have added a [BEM-like class name](https://meedan.atlassian.net/wiki/spaces/ENG/pages/1327628289/Naming+conventions+for+interactive+elements) to the element that is interacted with
- [ ] To the best of my knowledge, any new styles are applied according to the design system
- [ ] If I added a new external dependency, I included a rationale for doing so and an estimate of the change in bundle size (e.g., checked in https://bundlephobia.com/)
- [ ] If I touched a file that included an eslint-disable-file header, I updated the code such that the disabler can be removed

### CODE REVIEW

**Review by brianfleming — COMMENTED** [review]
very cool @sarahkeh I am looking forward to reviewing this.
Just wanted to put a note in that I pushed up some minor SVG optimizations that I have been meaning to give a tutorial on to the front-end team, but I have failed at doing that...coming soon to 2025! Just wanted to give you a heads up on my commit before I do a full review. 🙌

File: src/app/components/cds/inputs/ToggleButtonGroup.js [hunk 1]
```diff
@@ -85,6 +89,7 @@ ToggleButtonGroup.propTypes = {
   className: PropTypes.string,
   size: PropTypes.oneOf(['default', 'small', 'large']),
   helpContent: PropTypes.node,
+  iconCenter: PropTypes.bool,
```
Comment (amoedoamorim) [inline src/app/components/cds/inputs/ToggleButtonGroup.js:92]: Please also remove `/* eslint-disable react/sort-prop-types */` from this file

File: src/app/components/cds/inputs/ToggleButtonGroup.module.css [hunk 2]
```diff
@@ -49,6 +49,13 @@
         padding: 0 15px;
       }
     }
+
+    &.iconCenter {
+      :global(.MuiToggleButton-label) {
+        /* stylelint-disable-next-line scale-unlimited/declaration-strict-value */
+        font-size: 22px;
```
Comment (brianfleming) [inline src/app/components/cds/inputs/ToggleButtonGroup.module.css:56]: please change this to `font-size: 20px;`

File: src/app/components/cds/inputs/ToggleButtonGroup.module.css [hunk 3]
```diff
@@ -67,6 +74,13 @@
     :global(.MuiToggleButtonGroup-grouped) {
       padding: 0 10px;
     }
+
+    &.iconCenter {
+      :global(.MuiToggleButton-label) {
+        /* stylelint-disable-next-line scale-unlimited/declaration-strict-value */
+        font-size: 18px;
```
Comment (brianfleming) [inline src/app/components/cds/inputs/ToggleButtonGroup.module.css:81]: please change this to font-size: 16px;

File: src/app/components/cds/inputs/ToggleButtonGroup.module.css [hunk 4]
```diff
@@ -76,6 +90,13 @@
     :global(.MuiToggleButtonGroup-grouped) {
       padding: 0 20px;
     }
+
+    &.iconCenter {
+      :global(.MuiToggleButton-label) {
+        /* stylelint-disable-next-line scale-unlimited/declaration-strict-value */
+        font-size: 26px;
```
Comment (brianfleming) [inline src/app/components/cds/inputs/ToggleButtonGroup.module.css:97]: please change this to `font-size: 30px;`

**Review by brianfleming — CHANGES_REQUESTED** [review]
Couple things @sarahkeh and 100% all of them are my fault, so apologies! I did a poor job of spec-ing the visuals on this and forgot about one thing in the side rail that complicates this, but its pretty low stakes.

1. I left some inline comments about a few font-size changes to better match main button, and not the bad version i had in figma
2. with it having only two buttons, with the new light theme its a bit confusing that the hover background color is the same as the active state background color. So I think it would be best to: keep the bg color on hover of the inactive button to be `background-color: var(--color-beige-93);` and just let the icon color change
3. Since this is the 1st use of the toggle button group with icons AND vertical we need to tweak a few things about that layout of the buttons. if you look at the screenshot below (its a mockup to show whats going on), you will see that the horizontal button group is set to have the same height as the main button, so when we do layouts things can look aligned. but vertical, we need to match the width as well. So...what we should do is when `iconCenter` is used, the toggle button should be a square, just like the mainbutton
Screenshot 2024-12-19 at 10 53 08 AM

So as I mentioned, these are my fault for being lazy so I am happy to help with anything, just let me know if you would like me to.

**Review by amoedoamorim — APPROVED** [review]
Code LGTM. Please address Brian's requested changes

**Comment by sarahkeh** [comment]
@brianfleming Updated the styles like you suggested, lmk if it needs additional tweaks.  Also, thanks for updating the SVGs, normally I just delete the fill color in the <path> that figma adds, but when I did that with shield_check the whole check disappeared.

**Review by brianfleming — CHANGES_REQUESTED** [review]
Love it @sarahkeh looks great! thanks for those changes.
One last thing that I am seeing, I tried this on 2 different browsers, let me know if you can replicate:
On initial load of the page the tooltips don't show up until i click on the buttons, then after doing it once, they show up as expected. I don't understand whats going on. Take a look at this recording and see if this is happening for you:

https://github.com/user-attachments/assets/024e1a0d-f49d-4352-9781-cb55da5bd096

**Comment by sarahkeh** [comment]
@brianfleming That's so weird, I see it on my side, too.  I hope it's not a mysterious MaterialUI quirk.  I'll play around with it

**Comment by sarahkeh** [comment]
@brianfleming Figured it out! Needed to move the ToolTip into the button rather than around it, then wrap the contents in a <span>

**Review by brianfleming — APPROVED** [review]
so great @sarahkeh thanks for tracking down that tooltip issue so quickly!

dance

File: src/app/components/cds/inputs/ToggleButtonGroup.js [hunk 5]
```diff
@@ -85,6 +89,7 @@ ToggleButtonGroup.propTypes = {
...
   label: PropTypes.node,
   theme: PropTypes.oneOf(['default', 'setting']),
   variant: PropTypes.oneOf(['contained', 'outlined']),
```
Comment (brianfleming) [inline src/app/components/cds/inputs/ToggleButtonGroup.js:95]: @sarahkeh just noticed, shouldn't this include `containedLight`?

Comment (sarahkeh) [reply src/app/components/cds/inputs/ToggleButtonGroup.js]: Good catch, updated!

**Pull request merged**